kubectl/options.go

```diff
@@ -38,30 +54,73 @@ func (options *KubectlOptions) TempConfigFromAuthInfo() (string, error) {
 	}
 	logger.Infof("Created %s to act as temporary kubeconfig file.", tmpfile.Name())
 
+	scheme := options.AuthScheme()
+	switch scheme {
+	case DirectAuth:
+		err = tempConfigFromDirectAuthInfo(
+			logger, tmpfile, options.Server, options.Base64PEMCertificateAuthority, options.BearerToken)
+	case EKSClusterBased:
+		err = tempConfigFromEKSClusterInfo(logger, tmpfile, options.EKSClusterArn)
+	default:
+		return "", errors.WithStackTrace(AuthSchemeNotSupported{scheme})
+	}
+	return tmpfile.Name(), err
+
+}
+
+func tempConfigFromDirectAuthInfo(
+	logger *logrus.Entry,
+	tmpfile *os.File,
+	server string,
+	base64PEMCA string,
+	bearerToken string,
+) error {
 	config := &api.Config{}
-	err = AddClusterToConfig(
+	err := AddClusterToConfig(
 		config,
 		"default",
-		options.Server,
-		options.Base64PEMCertificateAuthority,
+		server,
+		base64PEMCA,
 	)
 	if err != nil {
-		return tmpfile.Name(), err
+		return err
 	}
 
 	logger.Infof("Adding auth info to config")
 	authInfo := api.NewAuthInfo()
-	authInfo.Token = options.BearerToken
+	authInfo.Token = bearerToken
 	config.AuthInfos["default"] = authInfo
 	logger.Infof("Done adding auth info to config")
 
-	err = AddContextToConfig(
+	return AddContextToConfig(
 		config,
 		"default",
 		"default",
 		"default",
 	)
-	return tmpfile.Name(), err
+}
+
+func tempConfigFromEKSClusterInfo(logger *logrus.Entry, tmpfile *os.File, eksClusterArn string) error {
+	server, b64PEMCA, token, err := getKubeCredentialsFromEKSCluster(eksClusterArn)
+	if err != nil {
+		return err
+	}
+	return tempConfigFromDirectAuthInfo(logger, tmpfile, server, b64PEMCA, token)
+}
+
+func getKubeCredentialsFromEKSCluster(eksClusterArn string) (string, string, string, error) {
```

Comment:
Is it a bit of a smell to return 4 values? Should this be a `struct` or a new type?

Reply:
Yup this was me being lazy. Will update.

Reply:
Done: 9966a97